### ISSUE TITLE
Copy.pm missing when running latexmk

### ISSUE THREAD

**Issue by koppor** [camilstaps/docker-texlive#1]
I'm trying to use the Dockerfile on TravisCI.

When running `docker run -v `pwd`:/var/tex camilstaps/texlive latexmk ausarbeitung.tex`, I get following output:

```
Can't locate File/Copy.pm in @INC (you may need to install the File::Copy module) (@INC contains: /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.20.2 /usr/local/share/perl/5.20.2 /usr/lib/x86_64-linux-gnu/perl5/5.20 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.20 /usr/share/perl/5.20 /usr/local/lib/site_perl .) at /usr/local/bin/latexmk line 129.
```

Full log at https://travis-ci.org/latextemplates/scientific-thesis-template/builds/243729259?utm_source=email&utm_medium=notification

**Comment by camilstaps** [comment]
I still had to update this to TL 2017, and believe this may be an artefact of that. Latexmk is working fine here now, could you check  it as well?

**Comment by koppor** [comment]
Travis pulled a new image, but still does not execute correctly: https://travis-ci.org/latextemplates/scientific-thesis-template/jobs/244607041#L98

**Comment by camilstaps** [comment]
Thanks.

This is very odd: I cloned your repository and attempted to build using the same commando as you give to Travis, and it works fine here. Also, if I inspect the container I can find the Copy module at `/usr/share/perl/5.20.2/File/Copy.pm`, and the `@INC` variable is set correctly.

I will investigate further, probably forking your repository and setting up Travis as well. Without reproducing the issue it's hard to solve it.

Thanks again.

**Comment by koppor** [comment]
Also happens with CircleCI 2.0: https://circleci.com/gh/latextemplates/scientific-thesis-template/10.

My fix was to build on [adnrv/texlive](https://hub.docker.com/r/adnrv/texlive/) and to install latexmk on top: https://github.com/koppor/docker-texlive/blob/master/Dockerfile

So, this issue is only "nice to have" to be solved, so I close it.

**Comment by camilstaps** [comment]
Good to hear, but I'd still like to keep it open to get it fixed. Sorry for not getting back to you, things have been rather hectic lately.

**Comment by djsutherland** [comment]
FYI, this is happening for me too, and I don't see `File/Copy.pm` present in the container:
```
$ docker run camilstaps/texlive ls /usr/share/perl/5.20.2/File
Basename.pm
Path.pm
Temp.pm
```

**Comment by koppor** [comment]
The solution for me was to create another image:
https://github.com/dante-ev/docker-texlive

Would be nice to know how you like it.

Dougal J. Sutherland <notifications@github.com> schrieb am Fr., 12. Okt.
2018, 17:42:

**Comment by camilstaps** [comment]
This has waited much too long, apologies. I committed an update which installs the `perl-modules` package in which `Copy.pm` can be found. I also updated to Debian Stretch. The file is now present:

```
$ docker run texlive ls /usr/share/perl/5.24.1/File
Basename.pm
Compare.pm
Copy.pm
Fetch.pm
Find.pm
GlobMapper.pm
Path.pm
Temp.pm
stat.pm
```

The image is still building on the Docker hub (https://hub.docker.com/r/camilstaps/texlive/builds/) and should be done in an hour or so.

@koppor I can now build `main-english.tex` of the current scientific-thesis-templates successfully with (though again, the image on the hub will be updated only in an hour or so):

```bash
docker run -v `pwd`:/var/tex camilstaps/texlive latexmk main-english.tex
```

@dougalsutherland I don't know what you were working on so cannot test it; if there are any issues I would be happy to help. (Now that I know how to find what packages provide files this fix was very simple and really didn't need to take this long.)
